tutorials/dataframe/df106_HiggsToFourLeptons.py

```diff
@@ -0,0 +1,194 @@
+## \file
+## \ingroup tutorial_dataframe
+## \notebook -draw
+## This tutorial is the Higgs to four lepton analysis from the ATLAS Open Data release in 2020
+## (http://opendata.atlas.cern/release/2020/documentation/). The data was taken with the ATLAS detector
+## during 2016 at a center-of-mass energy of 13 TeV. The decay of the Standard Model Higgs boson
+## to two Z bosons and subsequently to four leptons is called the "golden channel". The selection leads
+## to a narrow invariant mass peak on top a relatively smooth and small background, revealing
+## the Higgs at 125 GeV.
+##
+## The analysis is translated to a RDataFrame workflow processing about 300 MB of simulated events and data.
+##
+## \macro_image
+## \macro_code
+## \macro_output
+##
+## \date March 2020
+## \author Stefan Wunsch (KIT, CERN)
+
+import ROOT
+import json
+import os
+
+# Create a ROOT dataframe for each dataset
+# Note that we load the filenames from an external json file.
+path = "root://eospublic.cern.ch//eos/opendata/atlas/OutreachDatasets/2020-01-22"
+files = json.load(open(os.path.join(os.environ["ROOTSYS"], "tutorials/dataframe", "df106_HiggsToFourLeptons.json")))
+processes = files.keys()
+df = {}
+xsecs = {}
+sumws = {}
+samples = []
+for p in processes:
+    for d in files[p]:
+        # Construct the dataframes
+        folder = d[0] # Folder name
+        sample = d[1] # Sample name
+        xsecs[sample] = d[2] # Cross-section
+        sumws[sample] = d[3] # Sum of weights
+        samples.append(sample)
+        df[sample] = ROOT.RDataFrame("mini", "{}/4lep/{}/{}.4lep.root".format(path, folder, sample))
+
+# Select events for the analysis
+ROOT.gInterpreter.Declare("""
+using VecF_t = const ROOT::RVec<float>&;
+using VecI_t = const ROOT::RVec<int>&;
+bool GoodElectronOrMuon(VecI_t type, VecF_t pt, VecF_t eta, VecF_t phi, VecF_t e, VecF_t trackd0pv, VecF_t tracksigd0pv, VecF_t z0)
+{
+    for (size_t i = 0; i < type.size(); i++) {
+        ROOT::Math::PtEtaPhiEVector p(pt[i] / 1000.0, eta[i], phi[i], e[i] / 1000.0);
+        if (type[i] == 11) {
+            if (pt[i] < 7000 || abs(eta[i]) > 2.47 || abs(trackd0pv[i] / tracksigd0pv[i]) > 5 || abs(z0[i] * sin(p.Theta())) > 0.5) return false;
+        } else {
+            if (abs(trackd0pv[i] / tracksigd0pv[i]) > 5 || abs(z0[i] * sin(p.Theta())) > 0.5) return false;
+        }
+    }
+    return true;
+}
+""")
+
+for s in samples:
+    # Select electron or muon trigger
+    df[s] = df[s].Filter("trigE || trigM")
+
+    # Select events with exactly four good leptons conserving charge and lepton numbers
+    df[s] = df[s].Define("good_lep", "(abs(lep_eta) < 2.5) && (lep_pt > 5000) && (lep_ptcone30 / lep_pt < 0.3) && (lep_etcone20 / lep_pt < 0.3)")\
```

Comment:
When not realising that this is a mask, people might get confused. Maybe call it `good_lep_mask` or `good_lep_selection`?

Reply:
TBD :)

Reply:
Actually, I leave it like it is. We use this mechanic also in the vecops tutorials, so it's consistent like this.

Reply:
Hm, ok it was not the tutorials, it's the training material. Hmm ... I'm not sure what's best here, I see your point.

Reply:
I would also be in favour of `good_lep_mask`. It's a tutorial, doesn't hurt to err on the verbose side

Reply:
added a comment